### ISSUE TITLE
Uncaught ReferenceError: makeRichPromise is not defined

### ISSUE THREAD

**Issue by keikoro** [lektor/lektor#366]
Hiya, I'm (still) getting the error mentioned in #347 (not the original issue, but further down below in the thread) when using Lektor's admin. It does not only happen when pressing the "Reveal in Finder" button but, worse, also when trying to create a child page.

I'm using the development version of Lektor with Python 3.5.2 virtualenvs, in case it matters (though the errors look like they're related to JS). 

From Chrome's console:

```
utils.jsx?87e1:245 Uncaught ReferenceError: makeRichPromise is not defined
    at Object.apiRequest (eval at <anonymous> (http://127.0.0.1:5000/admin/static/gen/app.js:537:2), <anonymous>:242:12)
    at AddChildPage.createRecord (eval at <anonymous> (http://127.0.0.1:5000/admin/static/gen/app.js:692:2), <anonymous>:173:23)
    at Object.ReactErrorUtils.invokeGuardedCallback (eval at <anonymous> (http://127.0.0.1:5000/admin/static/gen/vendor.js:415:2), <anonymous>:70:16)
    at executeDispatch (eval at <anonymous> (http://127.0.0.1:5000/admin/static/gen/vendor.js:409:2), <anonymous>:85:21)
    at Object.executeDispatchesInOrder (eval at <anonymous> (http://127.0.0.1:5000/admin/static/gen/vendor.js:409:2), <anonymous>:108:5)
    at executeDispatchesAndRelease (eval at <anonymous> (http://127.0.0.1:5000/admin/static/gen/vendor.js:397:2), <anonymous>:43:22)
    at executeDispatchesAndReleaseTopLevel (eval at <anonymous> (http://127.0.0.1:5000/admin/static/gen/vendor.js:397:2), <anonymous>:54:10)
    at Array.forEach (native)
    at forEachAccumulated (eval at <anonymous> (http://127.0.0.1:5000/admin/static/gen/vendor.js:427:2), <anonymous>:24:9)
    at Object.processEventQueue (eval at <anonymous> (http://127.0.0.1:5000/admin/static/gen/vendor.js:397:2), <anonymous>:257:7)
```

**Comment by keikoro** [comment]
I'm also getting errors when clicking the Refresh Build button(s) and the Refresh Build dialogue never goes away (and cannot be closed by pressing "Cancel" either).

And when trying to save edits to an existing page, hmm.

**Comment by coreygrunewald** [comment]
Hi @kerstin,

I'm not able to reproduce the error you are seeing when attempting to create a child page. I can however produce an uncaught exception when enacting the `Refresh Build` dialog, but that doesn't hinder it's functionality. I'll be creating a PR for that issue.

Can you ensure that you are running the latest admin JS by doing the following:

1. Navigate to the `admin` directory of the lektor source
2. Run `rm -r node_modules; npm i; npm run webpack`
3. Restart your local server and see if you can still reproduce the error?

You should be able to view the source map versions of the JS modules in devtools - find `AddChildPage.jsx`. The `createRecord` should appear as such:

```
  createRecord () {
    const errMsg = (text) => {
      alert(i18n.trans('ERROR_PREFIX') + text)
    }

    const id = this.state.id || this.getImpliedId()
    if (!id) {
      errMsg(i18n.trans('ERROR_NO_ID_PROVIDED'))
      return
    }

    const data = {}
    const params = {id: id, path: this.getRecordPath(), data: data}
    if (!this.state.newChildInfo.implied_model) {
      data['_model'] = this.state.selectedModel
    }
    const primaryField = this.getPrimaryField()
    if (primaryField) {
      data[primaryField.name] = this.state.primary
    }

    utils.apiRequest('/newrecord', {json: params, method: 'POST'}, makeRichPromise)
      .then((resp) => {
        if (resp.exists) {
          errMsg(i18n.trans('ERROR_PAGE_ID_DUPLICATE').replace('%s', id))
        } else if (!resp.valid_id) {
          errMsg(i18n.trans('ERROR_INVALID_ID').replace('%s', id))
        } else {
          const urlPath = this.getUrlRecordPathWithAlt(resp.path)
          this.transitionToAdminPage('.edit', {path: urlPath})
        }
      })
  }
```

with a declaration of

```
import makeRichPromise from '../richPromise'
```

at the top of the file.

**Comment by keikoro** [comment]
@coreygrunewald Thanks, Corey, your instructions helped me fix my problem! There was no `node_modules` directory in my `admin` dir, but the `npm` commands were what was missing.

I just looked at the documentation again but couldn't find anything about those, so now I'm wondering if I keep looking in the wrong place, and why it worked fine the first time I installed Lektor but not when I reinstalled it.
